openwisp_ipam/api/views.py

```diff
@@ -165,24 +171,37 @@ class SubnetListCreateView(ListCreateAPIView):
     authentication_classes = (BearerAuthentication, SessionAuthentication)
     permission_classes = (DjangoModelPermissions,)
     pagination_class = ListViewPagination
-    queryset = Subnet.objects.all()
+
+    def get_queryset(self):
+        queryset = Subnet.objects.all()
+        user = self.request.user
+        if not user.is_superuser:
+            queryset = Subnet.objects.filter(organization__in=user.organizations_dict)
+        return queryset
 
 
 class SubnetView(RetrieveUpdateDestroyAPIView):
     serializer_class = SubnetSerializer
     authentication_classes = (BearerAuthentication, SessionAuthentication)
-    permission_classes = (DjangoModelPermissions,)
+    permission_classes = (
+        IsOrganizationMember,
+        DjangoModelPermissions,
+    )
     queryset = Subnet.objects.all()
 
 
 class IpAddressView(RetrieveUpdateDestroyAPIView):
     serializer_class = IpAddressSerializer
     authentication_classes = (BearerAuthentication, SessionAuthentication)
-    permission_classes = (DjangoModelPermissions,)
+    permission_classes = (
+        IsOrganizationMember,
+        DjangoModelPermissions,
+    )
     queryset = IpAddress.objects.all()
+    organization_field = 'subnet__organization'
 
 
-class RequestIPView(CreateAPIView):
+class RequestIPView(OrgPermissionMixin, CreateAPIView):
```

Comment:
Quick question, why can't we use 

```
    permission_classes = (IsOrganizationMember)
```

Here? Why are we using Mixins?

Reply:
@atb00ker When I try doing it with permission classes, 
```
permission_classes` = (IsOrganizationMember, DjangoModelPermissions,)
organization_field = 'subnet__organization'
```
it doesn't prevent user from other organizations from accessing it (look at the test I wrote below)
https://github.com/openwisp/openwisp-ipam/blob/3d6ae27e9e250ffff6ea96266a116fa987f8948b/openwisp_ipam/tests/test_multitenant.py#L200-L219
It gives 201 (Created) as a response. I couldn't figure out why this was happening, so I decided to create a mixin to fix this problem. If there is a better solution, I'd love to do that instead!

Reply:
I think we shouldn't need it, but let's get back to this after manual testing, check here: https://github.com/openwisp/openwisp-ipam/pull/76#pullrequestreview-532623227